### DISCUSSION TITLE
Enum naming convention in the source code

### DISCUSSION THREAD

**Discussion by kleonc** [Orama-Interactive/Pixelorama#433] [Q&A] [answered]
@OverloadedOrama, what's the preffered enum naming convention?
[Names of enums I've added recently](https://github.com/kleonc/Pixelorama/blob/92319bbcf05df08d2843eb7730b4ce6d90d54041/src/UI/TopMenuContainer.gd#L4-L8) are clearly wrong and I wanted to fix them but it seems that not all enums in the codebase follow the same convention. Both [`EnumName`](https://github.com/kleonc/Pixelorama/blob/92319bbcf05df08d2843eb7730b4ce6d90d54041/src/Autoload/Export.gd#L25-L27) and [`Enum_Name`](https://github.com/kleonc/Pixelorama/blob/92319bbcf05df08d2843eb7730b4ce6d90d54041/src/Autoload/Global.gd#L4-L8) are used. So which one is preffered?

**Comment by OverloadedOrama** [comment, marked as answer]
I recommend following the official [GDScript style guidelines](https://docs.godotengine.org/en/stable/getting_started/scripting/gdscript/gdscript_styleguide.html#constants-and-enums), so for enums, the name of the enum should be in PascalCase and the names of its members should be in CONSTANT_CASE.
